pallets/connectors/src/lib.rs

```diff
@@ -486,11 +552,139 @@ pub mod pallet {
 			Ok(())
 		}
 
+		/// Transfer non-tranche tokens to a given address
+		#[pallet::weight(< T as Config >::WeightInfo::transfer())]
+		#[pallet::call_index(7)]
+		pub fn transfer(
+			origin: OriginFor<T>,
+			currency_id: CurrencyIdOf<T>,
+			receiver: DomainAddress,
+			amount: <T as pallet::Config>::Balance,
+		) -> DispatchResult {
+			let who = ensure_signed(origin.clone())?;
+
+			ensure!(!amount.is_zero(), Error::<T>::InvalidTransferAmount);
+			// Check that the destination is not the local domain
+			ensure!(
+				receiver.domain() != Domain::Centrifuge,
+				Error::<T>::InvalidDomain
+			);
+			ensure!(
+				!CurrencyIdOf::<T>::is_tranche_token(currency_id),
+				Error::<T>::InvalidTransferCurrency
+			);
```

Comment:
That is actually a good reason for the trait. ^^ 

I think it makes sense to keep it and implement it for our currency id.